consts/src/lib.rs

```diff
@@ -172,6 +187,8 @@ impl Default for EdhocMessageBuffer {
 
 pub trait MessageBufferTrait {
     fn new() -> Self;
+    fn get(self, index: usize) -> Result<u8, AccessError>;
```

Comment:
The common pattern in the standard library is that fallible access just returns an Option (eg. the slice [get function](https://doc.rust-lang.org/std/primitive.slice.html#method.get)). Why deviate here?

Reply:
That is indeed an inconvenience. The means of making `?` just turn into a `Result<..., MyError>` are [not stable yet](https://doc.rust-lang.org/std/ops/trait.FromResidual.html) (if they even work that way). If you still want to take up the suggestion, there is a [pattern](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&code=pub+enum+Error+%7B%0A++++A%2C%0A++++B%2C%0A%7D%0A%0Atrait+BecomesErrorB+%7B%0A++++type+Out%3B%0A++++fn+or_b%28self%29+-%3E+Result%3CSelf%3A%3AOut%2C+Error%3E%3B%0A%7D%0A%0Aimpl%3CT%3E+BecomesErrorB+for+Option%3CT%3E+%7B%0A++++type+Out+%3D+T%3B%0A++++fn+or_b%28self%29+-%3E+Result%3CT%2C+Error%3E+%7B%0A++++++++self.ok_or%28Error%3A%3AB%29%0A++++%7D%0A%7D%0A%0Afn+returning_none%28%29+-%3E+Option%3Cu32%3E+%7B%0A++++None%0A%7D%0A%0Apub+fn+parse%28%29+-%3E+Result%3Cu32%2C+Error%3E+%7B%0A++++returning_none%28%29.or_b%28%29%3F%3B%0A++++Ok%280%29%0A%7D) that allows using `foo.get(10).or_parse_error()?` in the parsing code -- although it's really just a very small enhancement over simply writing `foo.get(10).ok_or(Error::ParseError)?` every time.

Reply:
I am working on an alternative parsing mechanism, inspired by [minicbor](https://crates.io/crates/minicbor), which should solve this in the parsing functions.